src/LiveComponent/src/Util/LiveControllerAttributesCreator.php

```diff
@@ -105,11 +105,11 @@ public function attributesForRendering(MountedComponent $mounted, ComponentMetad
             $attributesCollection->setRequestMethod($requestMethod);
         }
 
-        if ($liveMetadata->hasQueryStringBindings()) {
+        if ($liveMetadata->hasQueryStringBindings($mounted->getComponent())) {
             $queryMapping = [];
-            foreach ($liveMetadata->getAllLivePropsMetadata() as $livePropMetadata) {
+            foreach ($liveMetadata->getAllLivePropsMetadata($mounted->getComponent()) as $livePropMetadata) {
                 if ($livePropMetadata->queryStringMapping()) {
-                    $frontendName = $livePropMetadata->calculateFieldName($mounted, $livePropMetadata->getName());
+                    $frontendName = $livePropMetadata->calculateFieldName($mounted->getComponent(), $livePropMetadata->getName());
```

Comment:
I was confused by the (my) old code too. It's a bug that wasn't covered because I used a string field name in tests, and the component object is only used when the field name is computed by a method... Not directly related to this PR though, but I thought fixing the typo was worth it.

Reply:
I added another prop in test component to test with the method for field name